modules/OCPP201/conversions.cpp

```diff
@@ -0,0 +1,859 @@
+    case types::system::BootReason::Triggered:
+        return ocpp::v201::BootReasonEnum::Triggered;
+    case types::system::BootReason::Unknown:
+        return ocpp::v201::BootReasonEnum::Unknown;
+    case types::system::BootReason::Watchdog:
+        return ocpp::v201::BootReasonEnum::Watchdog;
+    default:
+        throw std::runtime_error("Could not convert BootReasonEnum");
+    }
+}
+
+ocpp::v201::ReasonEnum to_ocpp_reason(types::evse_manager::StopTransactionReason reason) {
+    switch (reason) {
+    case types::evse_manager::StopTransactionReason::DeAuthorized:
+        return ocpp::v201::ReasonEnum::DeAuthorized;
+    case types::evse_manager::StopTransactionReason::EmergencyStop:
+        return ocpp::v201::ReasonEnum::EmergencyStop;
+    case types::evse_manager::StopTransactionReason::EnergyLimitReached:
+        return ocpp::v201::ReasonEnum::EnergyLimitReached;
+    case types::evse_manager::StopTransactionReason::EVDisconnected:
+        return ocpp::v201::ReasonEnum::EVDisconnected;
+    case types::evse_manager::StopTransactionReason::GroundFault:
+        return ocpp::v201::ReasonEnum::GroundFault;
+    case types::evse_manager::StopTransactionReason::HardReset:
+        return ocpp::v201::ReasonEnum::ImmediateReset;
+    case types::evse_manager::StopTransactionReason::Local:
+        return ocpp::v201::ReasonEnum::Local;
+    case types::evse_manager::StopTransactionReason::LocalOutOfCredit:
+        return ocpp::v201::ReasonEnum::LocalOutOfCredit;
+    case types::evse_manager::StopTransactionReason::MasterPass:
+        return ocpp::v201::ReasonEnum::MasterPass;
+    case types::evse_manager::StopTransactionReason::Other:
+        return ocpp::v201::ReasonEnum::Other;
+    case types::evse_manager::StopTransactionReason::OvercurrentFault:
+        return ocpp::v201::ReasonEnum::OvercurrentFault;
+    case types::evse_manager::StopTransactionReason::PowerLoss:
+        return ocpp::v201::ReasonEnum::PowerLoss;
+    case types::evse_manager::StopTransactionReason::PowerQuality:
+        return ocpp::v201::ReasonEnum::PowerQuality;
+    case types::evse_manager::StopTransactionReason::Reboot:
+        return ocpp::v201::ReasonEnum::Reboot;
+    case types::evse_manager::StopTransactionReason::Remote:
+        return ocpp::v201::ReasonEnum::Remote;
+    case types::evse_manager::StopTransactionReason::SOCLimitReached:
+        return ocpp::v201::ReasonEnum::SOCLimitReached;
+    case types::evse_manager::StopTransactionReason::StoppedByEV:
+        return ocpp::v201::ReasonEnum::StoppedByEV;
+    case types::evse_manager::StopTransactionReason::TimeLimitReached:
+        return ocpp::v201::ReasonEnum::TimeLimitReached;
+    case types::evse_manager::StopTransactionReason::Timeout:
+        return ocpp::v201::ReasonEnum::Timeout;
+    default:
+        return ocpp::v201::ReasonEnum::Other;
+    }
+}
+
+ocpp::v201::ChargingStateEnum to_ocpp_charging_state_enum(types::evse_manager::StopTransactionReason reason) {
```

Comment:
That's a strange conversation, is that intentional?

Reply:
It was used to derive the Charging State from the StopTransactionReason for the TransactionEvent(Ended) event. But it looks like we have the charging state in libocpp internally, so we might want to remove this argument from `on_transaction_finished`.

Ill remove this function